src/main/java/org/jitsi/utils/dsi/DominantSpeakerIdentification.java

```diff
@@ -921,54 +920,42 @@ public DecisionMaker(DominantSpeakerIdentification algorithm)
         @Override
         public void run()
         {
-            try
-            {
-                do
-                {
-                    DominantSpeakerIdentification algorithm = this.algorithm.get();
+            boolean exit = false;
 
-                    if (algorithm == null)
-                    {
-                        break;
-                    }
-                    else
-                    {
-                        long sleep = algorithm.runInDecisionMaker(this);
+            DominantSpeakerIdentification<?> algorithm = this.algorithm.get();
 
-                        // A negative sleep value is explicitly supported i.e.
-                        // expected and is contracted to mean that this
-                        // DecisionMaker is instructed by the algorithm to
-                        // commit suicide.
-                        if (sleep < 0)
-                        {
-                            break;
-                        }
-                        else if (sleep > 0)
-                        {
-                            // Before sleeping, make the currentThread release
-                            // its reference to the associated
-                            // DominantSpeakerIdentification instance.
-                            algorithm = null;
-                            try
-                            {
-                                Thread.sleep(sleep);
-                            }
-                            catch (InterruptedException ie)
-                            {
-                                // Continue with the next iteration.
-                            }
-                        }
-                    }
+            if (algorithm == null)
+            {
+                exit = true;
+            }
+            else
+            {
+                long sleep = algorithm.runInDecisionMaker(this);
+
+                // A negative sleep value is explicitly supported i.e.
+                // expected and is contracted to mean that this
+                // DecisionMaker is instructed by the algorithm to
+                // commit suicide.
+                if (sleep < 0)
+                {
+                    exit = true;
+                }
+                else
+                {
+                    // Before sleeping, make the currentThread release
+                    // its reference to the associated
+                    // DominantSpeakerIdentification instance.
```

Comment:
I don't think this comment still applies?  It looks like it referred to the `algorithm = null` in the old code, which isn't relevant since the task isn't persisting.